src/script/scriptBuilder.zig

```diff
@@ -0,0 +1,239 @@
+const asBool = Script.asBool;
+const asInt = Script.asInt;
+const testing = std.testing;
+
+/// Maximum script length in bytes
+const MAX_SCRIPT_SIZE = 10000;
+
+/// ScriptBuilder Errors
+const ScriptBuilderError = error{ ScriptTooLong, ScriptTooShort };
+
+///ScriptBuilder is a library to generate easier scripts, useful for faster testing
+pub const ScriptBuilder = struct {
+    /// Dynamic array holding the opcodes
+    script: std.ArrayList(u8),
+
+    ///Memory Allocator
+    allocator: Allocator,
+
+    /// Initialize a new ScriptBuilder
+    ///
+    /// # Arguments
+    /// - `allocator`: Memory allocator for managing engine resources
+    ///
+    /// # Returns
+    /// - `!ScriptBuilder`: use with `try`
+    pub fn new(allocator: Allocator, capacity: usize) !ScriptBuilder {
+        return ScriptBuilder{
+            .script = try std.ArrayList(u8).initCapacity(allocator, capacity),
+            .allocator = allocator,
+        };
+    }
+
+    ///Push an OPcode to the ScriptBuilder
+    pub fn addOpcode(self: *ScriptBuilder, op: Opcode) !*ScriptBuilder {
+        if (self.script.items.len >= MAX_SCRIPT_SIZE) {
+            return ScriptBuilderError.ScriptTooLong;
+        }
+        try self.script.append(op.toBytes());
+        return self;
+    }
+
+    ///Push an Int to the SciptBuilder
+    pub fn addInt(self: *ScriptBuilder, num: i32) !*ScriptBuilder {
+        switch (num) {
+            0 => {
+                _ = try self.script.append(Opcode.OP_0.toBytes());
+            },
+            1...16 => try self.script.append(@intCast(Opcode.OP_1.toBytes() - 1 + num)),
+            else => {
+                _ = try self.addData(try ScriptNum.new(num).toBytes(self.allocator));
+            },
+        }
+        return self;
+    }
+
+    pub fn addData(self: *ScriptBuilder, data: []const u8) !*ScriptBuilder {
+        if (data.len == 0 or data.len == 1 and data[0] == 0) {
+            _ = try self.addOpcode(Opcode.OP_0);
+        } else if (data.len == 1 and data[0] <= 16) {
+            const op = Opcode.OP_1.toBytes() - 1 + data[0];
+            try self.script.append(op);
+        } else if (data.len == 1 and data[0] == 0x81) {
+            _ = try self.addOpcode(Opcode.OP_1NEGATE);
+        } else if (data.len < Opcode.OP_PUSHDATA1.toBytes()) {
+            _ = try self.script.append(@intCast(data.len));
+            for (data) |byte| {
+                try self.script.append(byte);
+            }
+        } else if (data.len >= Opcode.OP_PUSHDATA1.toBytes()) {
+            if (data.len > 75 and data.len <= 255) {
+                _ = try self.script.append(76);
+                const size = &[1]u8{@intCast(data.len)};
+                const n = std.mem.readInt(u8, size, .little);
+                _ = try self.script.append(n);
+
+                for (data) |byte| {
+                    try self.script.append(byte);
+                }
+            } else if (data.len >= 256 and data.len <= 65535) {
```

Comment:
`data.len >= 256` no necessary, your previous else condition make sure this is true. Remove it